### ISSUE TITLE
[Request] Option to add end-caps back to long notes

### ISSUE THREAD

**Issue by Buic** [seraxis/lr2oraja-endlessdream#21]
Would allow different note shapes (such as EZ2ON circles) to work without needing charge notes enabled.

Also, would be nice to have an additional option to make caps line up with the actual end of the note instead of being placed on top of it, if at all possible.

**Comment by MatVeiQaaa** [comment]
Placing the caps before the end, instead of at it, would create an issue of LN ends not lining up with other notes on the same line. I don't agree it's supposed to look like that.

**Comment by Buic** [comment]
> Placing the caps before the end, instead of at it, would create an issue of LN ends not lining up with other notes on the same line. I don't agree it's supposed to look like that.

Yeah that's why it would be optional; and it isn't the main request anyways.
Having an option for that would be more useful for note skins that aren't IIDX style, like these 

![image](https://github.com/seraxis/lr2oraja-endlessdream/assets/34635859/0555feee-152b-4c0a-8b7a-0cae5b73147c)
![image](https://github.com/seraxis/lr2oraja-endlessdream/assets/34635859/addb9a8e-5d39-417c-82b9-07875329198e)

**Comment by Marsoff9898** [comment]
Is it something called LN Mode?

**Comment by seraxis** [comment]
I'm not sure I understand, would this entail a new skin property and configuration option?

**Comment by Buic** [comment]
> I'm not sure I understand, would this entail a new skin property and configuration option?

As for supporting non-iidx style noteskins, that'd be a skin property; preferably a simple boolean that could easily be inserted into pre-existing skins to use those kinds of notes with them.

For having caps in LN mode I was thinking it'd just be a toggle in the launcher or mod menu; independent of skin. Perhaps skins should be able to force that on tho, since non-iidx notes would be too short without the ends rendering. 

I'm not really all that experienced with programming, but I think both things could probably be done by overriding or adding new checks to this function:

https://github.com/seraxis/lr2oraja-endlessdream/blob/8e8f66a9abaa6e6ece2b5bfd41fd0fd60add84bd/core/src/bms/player/beatoraja/play/LaneRenderer.java#L622

**Comment by MatVeiQaaa** [comment]
Why not have the both options as skin property? Having an option in launcher seems rather unnecessary, as you would likely only use that with certain skins made with that in mind.

**Comment by Buic** [comment]
Because it'd be more common to just want to turn caps back on for ppl who are used to how lr2 looks; I've seen requests for it come up a fair few times. The other thing is for more spesific styles of skin.

**Comment by seraxis** [comment]
I can look into this

**Comment by seraxis** [comment]
Implemented in #139
